### PR TITLE
Update DOI in citation instructions for 0.8.0 release

### PR DESCRIPTION
**Pull request by namurphy**
- [ ] I have added a changelog entry for this pull request.

- [ ] If adding new functionality, I have added tests and
      docstrings.

- [ ] I have fixed any newly failing tests.

### CODE REVIEW

File: docs/about/citation.rst [hunk 1]
```diff
@@ -9,15 +9,19 @@ specific version of PlasmaPy used in your project.  Citing a software
...
+.. |version_to_cite| replace:: 0.8.0
+.. |doi_for_version| replace:: 10.5281/zenodo.6774350
 
-   PlasmaPy Community et al. (2021). *PlasmaPy*, version 0.6.0, Zenodo,
-   https://doi.org/10.5281/zenodo.4602818
+Version |version_to_cite| of PlasmaPy may be cited with the following
+reference:
+
+   PlasmaPy Community et al. (2021). *PlasmaPy*, version |version_to_cite|,
+   Zenodo, https://doi.org/\ |version_to_cite|
```
Comment (namurphy) [inline docs/about/citation.rst:19]: - [ ] Double check that this works

**Review by rocco8773 — CHANGES_REQUESTED** [review]
Currently this does not work as intended.  Specifically the doi link mashing is not working.  I'm going to let you play with this, so ping me when it's ready for review.

Also, the replacements should either be moved to the top of file or put into `common_links.rst`.  The release guide would also need to be updated to reflect that these items would need updating during a release.

File: docs/about/citation.rst [hunk 2]
```diff
@@ -3,36 +3,43 @@
 Acknowledging and Citing
 ========================
 
-If you use PlasmaPy for a project that results in a research publication,
-we ask that you cite the `Zenodo <https://zenodo.org>`__ record for the
-specific version of PlasmaPy used in your project.  Citing a software
-package promotes scientific reproducibility, gives credit to its developers,
-and highlights the importance of software as a vital research product.
+.. |version_to_cite| replace:: 0.8.0
+.. |doi_hyperlink| replace:: https://doi.org/10.5281/zenodo.6774350
```
Comment (rocco8773) [inline docs/about/citation.rst:7]: When I go to this link I get a "DOI Not Found"?  Has this doi only been reserved and not yet "created"?

Comment (namurphy) [reply docs/about/citation.rst:7]: Correct: the link to the DOI does not exist yet, so this link won't work until after we've created the Zenodo record.  This does make it annoying to test, though...

Comment (rocco8773) [reply docs/about/citation.rst:7]: Ok, just wanted to double check before approving.

**Review by rocco8773 — APPROVED** [review]

**Pull request merged**